### PR TITLE
Matrix bug fixes and API test

### PR DESCRIPTION
**Pull request by HendrikLeuschner**
fixes issues of #63 and adds an api test for matrix

### CODE REVIEW

File: openrouteservice/src/main/java/heigit/ors/routing/algorithms/RPHASTAlgorithm.java [hunk 1]
```diff
@@ -275,19 +290,18 @@ private void fillEdgesUpward(MultiTreeSPEntry currEdge, PriorityQueue<MultiTreeS
...
-		//					upwardItersSkipped++;
+							//					upwardItersSkipped++;
 							continue;
 						}
+						_msptSubItem = ee.getItem(i);
 						
 						_tmpWeight = _edgeWeight + _entryWeight;
 
-						if (_msptSubItem.weight > _tmpWeight || _msptSubItem.weight == 0.0) {
+						if (_msptSubItem.weight > _tmpWeight || _msptSubItem.weight == Double.POSITIVE_INFINITY) {
```
Comment (karussell) [inline openrouteservice/src/main/java/heigit/ors/routing/algorithms/RPHASTAlgorithm.java]: I think just `_msptSubItem.weight > _tmpWeight` is necessary

File: openrouteservice/src/main/java/heigit/ors/routing/algorithms/RPHASTAlgorithm.java [hunk 2]
```diff
@@ -275,19 +290,18 @@ private void fillEdgesUpward(MultiTreeSPEntry currEdge, PriorityQueue<MultiTreeS
 						_msptItem = currEdge.getItem(i);
 						_entryWeight = _msptItem.weight;
 						
-						if (_entryWeight == 0.0)
+						if (_entryWeight == Double.POSITIVE_INFINITY)
```
Comment (karussell) [inline openrouteservice/src/main/java/heigit/ors/routing/algorithms/RPHASTAlgorithm.java:293]: You should use Double.isInfinity instead

Comment (HendrikLeuschner) [reply openrouteservice/src/main/java/heigit/ors/routing/algorithms/RPHASTAlgorithm.java:293]: If you look at the implementation of Double.isInfinite, it its
```public static boolean isInfinite(double var0) {
        return var0 == 1.0D / 0.0 || var0 == -1.0D / 0.0;
 }
```
of which 50% is the definition of Double.POSITIVE_INFINITY:
```
public static final double POSITIVE_INFINITY = 1.0D / 0.0;
```
As I do not want to accept Double.NEGATIVE_INFINITY, which would hint at an error in my code, I am happy to use my version, which should be faster as well.

File: openrouteservice/src/main/java/heigit/ors/routing/algorithms/RPHASTAlgorithm.java (hunk 1 above)
Comment (HendrikLeuschner) [reply openrouteservice/src/main/java/heigit/ors/routing/algorithms/RPHASTAlgorithm.java]: True!

File: openrouteservice-api-tests/src/test/java/heigit/ors/services/matrix/ResultsValidationTest.java [hunk 3]
```diff
@@ -89,95 +122,139 @@ public void emptyLocationEntryTest() {
...
+    private void checkTableDimensions(JSONObject json, String tableName, int rows, int columns) {
+        Assert.assertEquals(true, json.has(tableName));
+
+        JSONArray jTable = json.getJSONArray(tableName);
+        Assert.assertEquals(jTable.length(), rows);
+        Assert.assertEquals(jTable.getJSONArray(0).length(), columns);
+    }
+
+    @Test
+    public void distanceTest() {
```
Comment (rabidllama) [inline openrouteservice-api-tests/src/test/java/heigit/ors/services/matrix/ResultsValidationTest.java]: Can we write a bit of documentation about what the test is for?

File: openrouteservice-api-tests/src/test/java/heigit/ors/services/matrix/ResultsValidationTest.java [hunk 4]
```diff
@@ -89,95 +122,139 @@ public void emptyLocationEntryTest() {
...
+
+                JSONObject jResponseRouting = new JSONObject(response2.body().asString());
+                JSONObject jRoute = (jResponseRouting.getJSONArray("routes")).getJSONObject(0);
+                double routeDistance = jRoute.getJSONObject("summary").getDouble("distance");
+//                double matrixDistance;
+
+                double matrixDistance = jDistances.getJSONArray(i).getDouble(j);
+                Assert.assertTrue( matrixDistance - .1 < routeDistance);
+                Assert.assertTrue( matrixDistance + .1 > routeDistance);
+//                        .assertThat()
```
Comment (rabidllama) [inline openrouteservice-api-tests/src/test/java/heigit/ors/services/matrix/ResultsValidationTest.java]: We should try and make sure that we don't leave commented out code when we push to dev/master

File: openrouteservice-api-tests/src/test/java/heigit/ors/services/matrix/ResultsValidationTest.java [hunk 5]
```diff
@@ -21,6 +21,7 @@
 package heigit.ors.services.matrix;
 
 import static io.restassured.RestAssured.*;
+import static org.hamcrest.Matchers.*;
```
Comment (rabidllama) [inline openrouteservice-api-tests/src/test/java/heigit/ors/services/matrix/ResultsValidationTest.java]: Do any objects from this package get used? If not then it shouldn't be imported.

**Review by rabidllama — CHANGES_REQUESTED** [review]
Code seems OK, but we need to make sure that any new methods created which are complicated (especially tests) are well documented.

**Review by rabidllama — APPROVED** [review]

**Pull request merged**